src/Features/VisualBasic/Portable/CodeRefactorings/NodeSelectionHelpers.vb

```diff
@@ -2,24 +2,49 @@
 ' The .NET Foundation licenses this file to you under the MIT license.
 ' See the LICENSE file in the project root for more information.
 
+Imports System.Collections.Immutable
+Imports System.Threading
 Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.LanguageServices
 Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 
 Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings
     Friend Module NodeSelectionHelpers
-        Friend Async Function GetSelectedMemberDeclarationAsync(context As CodeRefactoringContext) As Task(Of SyntaxNode)
-            Dim methodMember = Await context.TryGetRelevantNodeAsync(Of MethodBaseSyntax)().ConfigureAwait(False)
-            If methodMember IsNot Nothing Then
-                Return methodMember
+        Friend Async Function GetSelectedMemberDeclarationAsync(context As CodeRefactoringContext) As Task(Of ImmutableArray(Of SyntaxNode))
+            Dim doc As Document = Nothing
+            Dim span As TextSpan = Nothing
+            Dim cancellationToken As CancellationToken = Nothing
+            context.Deconstruct(doc, span, cancellationToken)
```

Comment:
```suggestion
            Dim doc = context.Document;
            Dim span = context.Span;
            Dim cancellationToken = context.CancellationToken
```

VB basically doesn't benefit at all from deconstruction.  so better to just use the idiomatic assignment form above.